pkg/validation/validate_component.go

```diff
@@ -126,44 +160,46 @@ func (v *ComponentValidator) validateComponentsSet() []error {
 	// It is possible for there to be components that the component set does not
 	// know about, but all components in the component set must be in the
 	// components list
-	for _, ref := range v.componentSet.Spec.Components {
+	for i, ref := range v.componentSet.Spec.Components {
 		if _, ok := compMap[ref]; !ok {
-			errs = append(errs, fmt.Errorf("could not find component reference %v for any of the components", ref))
+			errs = append(errs, field.Duplicate(p.Child("Spec", "Components").Index(i), fmt.Sprintf("component ref %v", ref)))
```

Comment:
Should this be `Required` or `NotFound` instead of `Duplicate`?
Could also use `.Key(string)` here.

Reply:
Done